fl4health/clients/basic_client.py

```diff
@@ -394,38 +626,79 @@ def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
         """
         User defined method that returns a PyTorch Train DataLoader
         and a PyTorch Validation DataLoader
+
+        Args:
+            config (Config): The config from the server.
+
+        Returns:
+            Tuple[DataLoader, DataLoader]: The client train and validation loader.
+
+        Raises:
+            NotImplementedError: To be defined in child class.
         """
         raise NotImplementedError
 
     def get_criterion(self, config: Config) -> _Loss:
         """
         User defined method that returns PyTorch loss to train model.
+
+        Args:
+            config (Config): The config from the server.
+
+        Raises:
+            NotImplementedError: To be defined in child class.
         """
         raise NotImplementedError
 
     def get_optimizer(self, config: Config) -> Union[Optimizer, Dict[str, Optimizer]]:
         """
         Method to be defined by user that returns the PyTorch optimizer used to train models locally
         Return value can be a single torch optimizer or a dictionary of string and torch optimizer.
```

Comment:
Maybe we can mention a situations where we might have multiple optimizers. I.e. APFL?